### DISCUSSION TITLE
preventing bot spamming

### DISCUSSION THREAD

**Discussion by rduToNature** [emailjs-com/help#47] [Q&A] [answered]
Hi,

According to the official documentation there should be an ip based bot prevention design, but i just tried to simulate bot attack with puppeteer on the production site and I can sends hundreds of e-mail without being stopped. is the rate limit being set to high or do I have to enable this measure?

**Comment by xr0master** [comment, marked as answer]
Hi, the limit rate is 1 request per second.

**Reply by rduToNature** [reply]
Thanks, is there anyway I can change it? Make it something like 5 request per hour per ip.

**Reply by xr0master** [reply]
Unfortunately, we do not have such a feature yet. We are thinking in the future to add the ability to set up a rate limit for each template, but this will not be done soon.

**Reply by hardik-rajpal** [reply]
This seems like a core issue with a service whose main purpose is to facilitate client-side requests to trigger emails. The only reason it's not permitted by the usual architecture is the potential DoS attacks.

**Reply by xr0master** [reply]
> This seems like a core issue with a service whose main purpose is to facilitate client-side requests to trigger emails. The only reason it's not permitted by the usual architecture is the potential DoS attacks.

How many times have you been attacked already?

**Reply by hardik-rajpal** [reply]
I haven't been attacked so far. The only reason I can't use this is because the website I'm designing is for a cybersecurity firm. I can't risk using this for fear of the headlines an attack would make. I love the package otherwise.
